src/main/java/io/appium/java_client/pagefactory/README.md

```diff
@@ -0,0 +1,75 @@
+# Standard Selectors
+
+## AndroidFindBy / iOSXCUITFindBy / WindowsFindBy
+
+# Advanced Selectors
+
+## iOS's Class Chain Queries
+
+Our XCUiTest integration has full support for the 'Class Chain' concept. This
+can do much of what XPath does...but faster.
+
+### External References
+
+Refer to [the official WebDriverAgent query docs](https://github.com/facebookarchive/WebDriverAgent/wiki/Class-Chain-Queries-Construction-Rules)
+to learn more about the general concept.
+
+### Sample usage:
+
+```java
+// Selector for image elements
+@iOSXCUITFindBy(iOSClassChain = "**/XCUIElementTypeImage")
+
+// Selector for every cell with the name 'Foo'
+@iOSXCUITFindBy(iOSClassChain = "**/XCUIElementTypeCell[`name == \"Foo\"`]")
+// Selector for every cell with a name that starts with 'Foo'
+@iOSXCUITFindBy(iOSClassChain = "**/XCUIElementTypeCell[`name BEGINSWITH \"Foo\"`]")
+
+// Selector that'll match every top-level element on screen
+@iOSXCUITFindBy(iOSClassChain = "*")
+// Selector that'll match every leaf element on screen (watch out: this can be SLOW)
+@iOSXCUITFindBy(iOSClassChain = "**/*")
+```
+
+## Android's uiAutomator String
+
+Available when using [AndroidBy](AndroidBy) and [AndroidFindBy](AndroidFindBy) with
+[appium-uiautomator2-server](https://github.com/appium/appium-uiautomator2-server). This
+string will be used by the server to construct a UiSelector or UiScrollable object.
+
+### External References
+
+For an overview of what the backend is capable of, please check out the
+
+* [Main UI Automator Guide](https://developer.android.com/training/testing/ui-automator)
+* [UiScrollable API docs](https://developer.android.com/reference/androidx/test/uiautomator/UiScrollable)
+and
+* [UiSelector API docs](https://developer.android.com/reference/androidx/test/uiautomator/UiSelector)
+
+### Sample Strings
+
+Here are some ways you could configure a UiSelector in your project:
+
+```java
+// Create a selector that looks for the text "Hello World":
+@AndroidFindBy(uiAutomator = "new UiSelector().text(\"Hello World\")")
```

Comment:
I can observe that people are mostly looking for methods to scroll into views, like https://discuss.appium.io/t/cant-get-setmaxsearchswipes-to-work-with-driver-findelement-mobileby-androiduiautomator/29935

It would be nice to put some examples of such locators here as well

Reply:
Right - I'll need to read up on that.